pkg/sqlite/migrator.go

```diff
@@ -0,0 +1,122 @@
+package sqlite
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database/sqlite3"
+	_ "github.com/golang-migrate/migrate/v4/source/file" // indirect import required by golang-migrate package
+)
+
+const (
+	// Hardcoded path where the db_migrations live
+	defaultMigrationsPath = "./pkg/sqlite/db_migrations"
+)
+
+type SQLMigrator struct {
+	db             *sql.DB
+	migrationsPath string
+}
+
+// NewSQLLiteMigrator returns a SQLMigrator. The SQLMigrator takes a sql database and directory for migrations
+// and exposes a set of functions that allow the golang-migrate project to apply migrations to that database.
+func NewSQLLiteMigrator(db *sql.DB, migrationsPath string) *SQLMigrator {
+	if migrationsPath == "" {
+		migrationsPath = defaultMigrationsPath
+	}
+	return &SQLMigrator{
+		db:             db,
+		migrationsPath: migrationsPath,
+	}
+}
+
+// InitMigrationVersion parses the db_migrations for the latest migration version, then applies that
+// version manually to the database after the DB in initialized. This is to ensure that the correct
+// future db migration will be run once the database is built.
+//
+// This function does not take any input and relies on the db defined in the SQLMigrator object
+// to connect to the database.
+func (m *SQLMigrator) InitMigrationVersion() error {
+	instance, err := sqlite3.WithInstance(m.db, &sqlite3.Config{})
+	if err != nil {
+		return err
+	}
+
+	migrator, err := migrate.NewWithDatabaseInstance(fmt.Sprintf("file://%s", m.migrationsPath), "registrydb", instance)
```

Comment:
Do we need the database name for anything anywhere else?

Reply:
That name exists purely for logging purposes per the API documentation:

https://godoc.org/github.com/golang-migrate/migrate#NewWithDatabaseInstance

It gets attached to the migration object when performing updates internally since the only other identifier is the name of the db file and gets thrown away afterwards.